### DISCUSSION TITLE
Custom Route response returns as 404.

### DISCUSSION THREAD

**Discussion by YussufElarif** [strapi/strapi#6554] [Legacy Topics]
**Describe the bug**
After creating a custom route in strapi, it successfully hits the endpoint and sends a response back to the client. However, it seems that the response in ctx is returning as 404 and therefore cannot retrieve what is inside the body.

The screenshots below will show you the only custom changes that I've made to the backend

**Steps to reproduce the behavior**

1. After creating the custom route, send a Graphql or a normal post request
2. console.log the ctx.request

**Screenshots**
api/order/controllers/order.js
![image](https://user-images.githubusercontent.com/9627828/83809608-a4586c00-a6ae-11ea-878e-be2974b1f822.png)

api/order/config/routes.json (no custom changes made here)
![image](https://user-images.githubusercontent.com/9627828/83808866-66a71380-a6ad-11ea-986d-ccaaae0ccf58.png)

api/order/config/schema.graphql.js
![image](https://user-images.githubusercontent.com/9627828/83808929-80485b00-a6ad-11ea-92f3-016fdd378749.png)

After making a request, it returns fine (GraphQL).
![image](https://user-images.githubusercontent.com/9627828/83809766-eda8bb80-a6ae-11ea-8d2a-ffffc7562fe0.png)

After making a request (normal POST)
![image](https://user-images.githubusercontent.com/9627828/83809817-01ecb880-a6af-11ea-9678-edb501cc416c.png)

but the console log shows the body as undefined for both requests...
![image](https://user-images.githubusercontent.com/9627828/83809310-1ed4bc00-a6ae-11ea-9b87-5a6f1dca05e5.png)

**System**

- Node.js version:  12.16.3 
- NPM version: 6.14.4
- Strapi version: 
- Database: bookshelf/sqlite
- Operating system: Windows

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.

**Reply by YussufElarif** [reply]
Initially, i thought it was a bug but it turns out that i had just used the wrong property. I used 'response' instead of 'request'. I'm happy to close this issue.

**Comment by elliottross23** [comment]
+1 I'm getting a 404 in a custom route but there's no error in the console. I even wrapped the custom handler in a try/catch and can't get anywhere...

**Reply by elliottross23** [reply]
I think strapi is suppressing errors and it's making it difficult to debug....
